design-spec/priviledge-decentralized-swupdates.tex

```diff
@@ -193,8 +245,92 @@ \section{Satisfying the requirements}
+vote-transactions pay are calculated in the same way as monetary transactions.
+In this way, stakeholders can vote directly by submitting a vote, or indirectly
+by delegating their voting rights to experts.
+
+\subsection{Democratic}
+\label{sec:sat-decentr-decis-making}
+
+This design addresses the requirements that the community can vote on updates,
+by defining an on-chain voting mechanism in which participants either:
+\begin{itemize}
+\item vote directly by submitting a vote to the blockchain in the same way they
+  do for regular transactions, or
+\item delegate their vote to experts who will vote on their behalf.
+\end{itemize}
+
+\subsection{Protocol driven}
+\label{sec:sat-protocol-driven}
+
+The update mechanism here described can be run as part of the ledger rules of a
+blockchain.
+
+\subsection{Transparent and auditable}
+\label{sec:sat-transp-audit}
+
+By making the update proposals and votes part of the transactions that are
+included in blocks, we ensure that the system updates history is stored in the
+immutable ledger that is the blockchain.
+
+\subsection{Secure}
+\label{sec:sat-secure}
+
+% TODO: is delegation to experts part of the security of the protocol?
+% Delegation to experts:
+%
+% - makes participation more likely (although this depends on the experts' incentives)
+%
+% - put's important decisions in the hands of the people that have the technical
+%   knowledge
+%
+
+\subsection{Performant and scalable}
+\label{sec:sat-performant-scalable}
+
+Our back-of-the envelope calculations showed that the update protocol has an
+impact on the system throughput that is linear on the number of participants.
+These numbers were confirmed by the measurements we took after running several
+experiments on a testnet.
+
+Our asymptotic complexity analysis and micro-benchmarks showed that the impact
+on system's performance is also linear on the number of participants.
+
+\subsection{Metadata-driven}
+\label{sec:sat-metadata-driven}
+
+We have define an update protocol where the submitter of a proposal can specify
+(when applicable):
+\begin{itemize}
+\item its priority,
+\item its dependency,
+\item the duration of its voting period,
+\item the duration of its deployment window.
+\end{itemize}
+
+\subsection{Consistent}
+\label{sec:sat-cons-update-logic}
+
+The activation protocol in Section~\ref{sec:activation} defines a queuing
+mechanism that ensures that proposals are activated in the order prescribed by
+their priorities. Furthermore, this queuing mechanism ensures a candidate
+for activation always supersedes the current protocol version.
+
+The voting period of a proposal submitted to either the ideation or approval
+phases is taken from the proposal's metadata.
+% TODO: we should make sure that we cap this voting period, and mention this
+% exception here.
+Similarly, the deployment window, in the case of consensus-impacting updates, is
+also determined by the proposal's metadata.
```

Comment:
Not only that, we also handle conflicting proposal which supersede the same version and also provide an emergency mechanism for cancelling proposals who are being endorsed upon the approval of the community.